cpp/arcticdb/stream/merge_utils.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <arcticdb/column_store/chunked_buffer.hpp>
+#include <arcticdb/column_store/string_pool.hpp>
+#include <arcticdb/column_store/memory_segment.hpp>
+
+namespace arcticdb {
+inline void merge_string_column(
+    ChunkedBuffer& src_buffer,
+    const std::shared_ptr<StringPool>& src_pool,
+    const std::shared_ptr<StringPool>& merged_pool,
+    CursoredBuffer<ChunkedBuffer>& output,
+    bool verify
+) {
+    using OffsetType = StringPool::offset_t;
+    constexpr auto offset_size =  sizeof(OffsetType);
+    auto num_strings = src_buffer.bytes() / offset_size;
+    for(auto row = size_t(0); row < num_strings; ++row) {
```

Comment:
NIT: having both `auto` and the actual types seems redundant. We could just type `size_t row = 0`

Reply:
Think I relocated this code from elsewhere, I agree it is a bit odd. Although if we are initializing with literal types I would prefer to be specific, as the above actually says 'unsigned long row = int'